make/Makefile.mettle

```diff
@@ -61,7 +61,6 @@ $(BUILD)/mettle/Makefile: $(TOOLS) $(ROOT)/mettle/configure \
 	@echo "Configuring mettle for $(TARGET)"
 	@mkdir -p $(BUILD)/mettle
 	@$(SETUP_BUILDENV) cd $(BUILD)/mettle; \
```

Comment:
```suggestion
	@$(SETUP_BUILDENV) cd $(ROOT)/mettle; \
		$(CONFIG_FIXUP) \
		cd $(BUILD)/mettle; \
```

Leaving `CONFIG_FIXUP` in place instead of adding duplicate versions of `config.guess` and `config.sub` to the mettle directory would be easier to maintain consistency.

```
vagrant@osx-11 mettle % diff deps/config.sub mettle/config.sub
vagrant@osx-11 mettle % diff deps/config.guess mettle/config.guess
```

Another option (maybe even better) might be to reverse the location here and instead of maintaining these files in `deps` and update `CONFIG_FIXUP` to copy them from the mettle path.

```
diff --git a/make/Makefile.mettle b/make/Makefile.mettle
index 575e1e1..64c4be4 100644
diff --git a/make/Makefile.common b/make/Makefile.common
index f81f4e0..3ceae36 100644
--- a/make/Makefile.common
+++ b/make/Makefile.common
@@ -26,7 +26,7 @@ ifeq (,$(findstring mingw,$(TARGET)))
 endif

 CPPFLAGS:=$(CPPFLAGS) -I$(BUILD)/include -DCURL_STATICLIB
-CONFIG_FIXUP=cp $(DEPS)/config.guess $(DEPS)/config.sub . ;
+CONFIG_FIXUP=cp $(ROOT)/mettle/config.guess $(ROOT)/mettle/config.sub . ;
 CONFIGURE=configure --prefix=$(BUILD) --disable-shared ac_cv_path_PKGCONFIG=$(CWD)/pkg-config
 CONFIGURE_STATIC=configure --prefix=$(BUILD) --static
 ifneq "$(TARGET)" "native"
```

Reply:
> Another option (maybe even better) might be to reverse the location here and instead of maintaining these files in `deps` and update `CONFIG_FIXUP` to copy them from the mettle path.
> 
> ```
> diff --git a/make/Makefile.mettle b/make/Makefile.mettle
> index 575e1e1..64c4be4 100644
> diff --git a/make/Makefile.common b/make/Makefile.common
> index f81f4e0..3ceae36 100644
> --- a/make/Makefile.common
> +++ b/make/Makefile.common
> @@ -26,7 +26,7 @@ ifeq (,$(findstring mingw,$(TARGET)))
>  endif
> 
>  CPPFLAGS:=$(CPPFLAGS) -I$(BUILD)/include -DCURL_STATICLIB
> -CONFIG_FIXUP=cp $(DEPS)/config.guess $(DEPS)/config.sub . ;
> +CONFIG_FIXUP=cp $(ROOT)/mettle/config.guess $(ROOT)/mettle/config.sub . ;
>  CONFIGURE=configure --prefix=$(BUILD) --disable-shared ac_cv_path_PKGCONFIG=$(CWD)/pkg-config
>  CONFIGURE_STATIC=configure --prefix=$(BUILD) --static
>  ifneq "$(TARGET)" "native"
> ```

Yeah that works as well, its the same thing though. I'll update the code with this change since it seems neater.